src/components/file-preview.js

```diff
@@ -0,0 +1,214 @@
+import PropTypes from 'prop-types';
+import prettyBytes from 'pretty-bytes';
+import MiniPager from './mini-pager.js';
+import { message, Tooltip } from 'antd';
+import Logger from '../common/logger.js';
+import MediaPlayer from './media-player.js';
+import { useTranslation } from 'react-i18next';
+import LiteralUrlLink from './literal-url-link.js';
+import { useService } from './container-context.js';
+import { handleError } from '../ui/error-helper.js';
+import { useDateFormat } from './locale-context.js';
+import mimeTypeHelper from '../ui/mime-type-helper.js';
+import { RESOURCE_TYPE } from '../domain/constants.js';
+import ClientConfig from '../bootstrap/client-config.js';
+import { getResourceType } from '../utils/resource-utils.js';
+import { FileOutlined, FileTextOutlined } from '@ant-design/icons';
+import CopyToClipboardIcon from './icons/general/copy-to-clipboard-icon.js';
+import React, { Fragment, useEffect, useMemo, useRef, useState } from 'react';
+import PdfDocument, { PDF_DOCUMENT_STRETCH_DIRECTION } from './pdf-document.js';
+
+const logger = new Logger(import.meta.url);
+
+function FilePreview({ lastModified, size, url }) {
```

Comment:
Maybe `lastModified` could rather be `createdOn`, cause the label used to display it is `"common:createdOn"`.
Or maybe `uploadedOn` would actually be even better (with a corresponding label).

Reply:
Changed. The reason I used `lastModified` was that it is what we get from the server in the file object. But I find `createdOn` better, too (as long as we consider overriding a file with the same name in private storage a new "creation"). We should then change the server response accordingly once we got rid of the old browser.

Reply:
👍